### PR TITLE
Fail-safes for stamina drop in pulping and butchering

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Balance "Fail safes for stamina drop in pulping an butchering"```

#### Purpose of change
Addresses some issues in ##31889
Mainly I can certainly agree with the idea of a logical player character taking a rest during long activities and saving some breath for later, just in case.

#### Describe the solution
So:
* pulping stamina drain was reduced to half pulping power of a player.
* added fail safe to pulping so no risk of winded effect, ever; butchering already had this
* in pulping an butchering player "reserves" 1/3 of max stamina, and rests during the activity - activity takes longer but character controls his stamina in-action

#### Describe alternatives you've considered
Perhaps I'll add an option to choose how big the reserve should be in percentage in the future.

#### Additional context
I may return here for a deeper review of the code, but I'll be away for some time so this has to suffice for now.

### CODE REVIEW

**Comment by kevingranade** [comment]
Regarding #31915 there is a very valid point in there, which is that non-combat activities should not trigger the winded state ever. (impact of the winded state is a separate issue we should address elsewhere)

As such, I think it would be reasonable to move handling of this outside the individual activity handlers and instead handle them at the top level for all activities.  If there's an exception where it does make sense to trigger the winded state for a long activity, we can still make that configurable in the activity definition.

So basically we'd have one chunk of code monitoring stamina, and if too low it would trigger a "rest" and consume some moves instead of proceeding directly to the activity, so we'd get a stamina buffer across all activities.

**Comment by nexusmrsep** [comment]
Closing, because #31951 does this job on the higher level.
